src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -344,37 +335,54 @@ protected unsafe void UnsafeWriteUnicodeChars(char* chars, int charCount)
 
         protected unsafe int UnsafeGetUnicodeChars(char* chars, int charCount, byte[] buffer, int offset)
         {
-            char* charsMax = chars + charCount;
-            while (chars < charsMax)
+            if (BitConverter.IsLittleEndian)
             {
-                char value = *chars++;
-                buffer[offset++] = (byte)value;
-                value >>= 8;
-                buffer[offset++] = (byte)value;
+                new ReadOnlySpan<byte>((byte*)chars, 2 * charCount)
+                    .CopyTo(buffer.AsSpan(offset));
             }
+            else
+            {
+                char* charsMax = chars + charCount;
+                while (chars < charsMax)
+                {
+                    char value = *chars++;
+                    buffer[offset++] = (byte)value;
+                    buffer[offset++] = (byte)(value >> 8);
+                }
+            }
+
             return charCount * 2;
         }
 
         protected unsafe int UnsafeGetUTF8Length(char* chars, int charCount)
         {
             char* charsMax = chars + charCount;
-            while (chars < charsMax)
+
+            // This method is only called from 2 places and will use length of at least (128/3 and 256/3) respectivly
+            // We avoid Vector<T> sine it is unsure how downclocking due to AVX512 would affect total throughput
+            if (Vector256.IsHardwareAccelerated
+                && Vector256<short>.Count < charCount && charCount <= 2048)
```

Comment:
`Vector<T>` doesn't support AVX512 today.  And just using `Vector256<T>` means there won't be any vectorization on Arm64.

Reply:
The original version was just Vector but with a check for at least 256 width. I don't remember the numbers but I think having just Vector128 made the cutoff for selecting system.encoding much lower and it might be better to just call System.encoding directly. 

Without having any measurements for arm that approach seems safest to me since it will still be faster than the current code.

Reply:
> that approach

Which approach?  I'm not sure which "that" refers to.

I've not seen a demonstrable need for performance improvements in this code.  While we of course welcome perf improvements, if they come at a maintainability expense, we want those changes to be worth it.  In general, we want folks using `Encoding.UTF8.GetBytes` and not hand-rolling UTF8 encoding implementations, as that's both the most maintainable and correct, and should also yield very good performance across platforms.

Unless there's a significant performance advantage to not using `Encoding.UTF8.GetBytes` and a demonstrable need for that improvement in a real-world context, my strong preference is to just use `Encoding.UTF8` and not try to manually vectorize a hand-rolled replacement in a corner of the libraries.

Reply:
> Which approach? I'm not sure which "that" refers to.
I think i reworded the commented on the phone before I saw your reploy.

> Unless there's a significant performance advantage to not using Encoding.UTF8.GetBytes and a demonstrable need for that improvement in a real-world context, my strong preference is to just use Encoding.UTF8 and not try to manually vectorize a hand-rolled replacement in a corner of the libraries.

Thank you that was exactly the kind of feedback I was after in question 2 under "Feedback wanted" 👍 
While the UnsafeGetBytes did give some nice increase in RPS for xml based web services i have not had any specific case  with very long text so it does make any big difference to me.

Reply:
Based on feedback I changed apprach to "2b. Just call Encoding.GetByteCount"